frappe/desk/listview.py

```diff
@@ -26,8 +26,14 @@ def set_list_settings(doctype, values):
 	doc.save()
 
 @frappe.whitelist()
-def get_user_assignments_and_count():
-	user_list = frappe.get_list("User", filters={"user_type": "System User"})
-	assignment_data = sorted([{"count":frappe.db.count('ToDo', filters = {'reference_type': 'Issue', 'owner': user['name'], 'status': 'Open'}),
+def get_user_assignments_and_count(doctype, current_filters):
+	filtered_list = frappe.get_all(doctype,
+		filters=current_filters)
+	names = list(map(lambda x: x.name, filtered_list))
+	user_list = frappe.get_all("User", filters={"user_type": "System User"})
+	if names:
+		assignment_data = sorted([{"count":frappe.db.count('ToDo', filters = {'owner': user['name'], 'status': 'Open', 'reference_name': ('in', names)}),
```

Comment:
Instead of querying for each user in user_list, query once for all users, and then group them in python.